### PR TITLE
use nodejs in-built support for es modules

### PR DESCRIPTION
**Pull request by Sanddrah**
Hi there.
I've updated the package to use nodejs' in-built support for es modules. So no need for esm dependency.
Accepting this update I am proposing will help us in maintaining the package in Debian so we don't have to keep too many patches.
Thanks!

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ import * as rollup from 'rollup';
   if (result_es5.emitSkipped) {
     const diagnostic = result_es5.diagnostics[0];
     const error = new Error(diagnostic.messageText);
-    error.stack = error.stack.replace(/at .*/, 'at ' + diagnostic.file.fileName + ':' + diagnostic.file.lineMap.findIndex(value => value > diagnostic.start));
+    //error.stack = error.stack.replace(/at .*/, 'at ' + diagnostic.file.fileName + ':' + diagnostic.file.lineMap.findIndex(value => value > diagnostic.start));
+    console.error(diagnostic);
```
Comment (twiss) [inline build.js:27]: It's not super important but could you either update the stack in a way that works (if it's the line map that's problematic, maybe just remove that part?) or remove that line altogether, instead of just commenting it out?

File: src/bignum/bignum.ts [hunk 2]
```diff
@@ -10,17 +10,7 @@ export const _bigint_stdlib = { Uint32Array: Uint32Array, Math: Math };
...
-}
-
-if (_bigint_stdlib.Math.imul === undefined) {
-  _bigint_stdlib.Math.imul = _half_imul;
-  _bigint_asm = bigint_asm(_bigint_stdlib, null, _bigint_heap.buffer);
-  delete _bigint_stdlib.Math.imul;
-} else {
-  _bigint_asm = bigint_asm(_bigint_stdlib, null, _bigint_heap.buffer);
-}
+_bigint_asm = bigint_asm(_bigint_stdlib, null, _bigint_heap.buffer);
```
Comment (twiss) [inline src/bignum/bignum.ts:13]: I think this could just be
```ts
export const _bigint_asm = bigint_asm(_bigint_stdlib, null, _bigint_heap.buffer);
```
(and then we probably also don't have to import `bigintresult` above)

File: build.js [hunk 3]
```diff
@@ -96,8 +97,8 @@ import * as rollup from 'rollup';
     },
   });
   await es8bundle.write({
-    file: 'asmcrypto.all.es8.js',
-    format: 'es',
+    file: 'asmcrypto.all.es8.cjs',
+    format: 'cjs',
   });
```
Comment (twiss) [inline build.js:102]: Why is this needed, if we want to use Node.js's built-in support for ES modules? :sweat_smile:

File: src/bignum/bignum.ts (hunk 2 above)
Comment (Sanddrah) [reply src/bignum/bignum.ts:13]: Thanks @twiss! I'll make the necessary changes

File: build.js (hunk 3 above)
Comment (Sanddrah) [reply build.js:102]: Noted :)

**Comment by larabr** [comment]
Superseded by #8. Thanks for initiating work on this :)
